querybook/webapp/ui/Comment/Comments.tsx

```diff
@@ -0,0 +1,257 @@
+        } else {
+            // TODO: save comment
+        }
+        setCurrentComment(initialCommentValue);
+    }, [editingCommentId]);
+
+    const onEnterPress = React.useCallback(
+        (evt: KeyboardEvent) => {
+            if (
+                matchKeyMap(evt, {
+                    key: 'Enter',
+                    name: 'Comment',
+                })
+            ) {
+                handleCommentSave();
+            }
+        },
+        [handleCommentSave]
+    );
+
+    useEvent('keydown', onEnterPress);
+
+    const renderFlatCommentDOM = (comment: IComment) => (
+        <Comment
+            key={comment.id}
+            text={DraftJs.ContentState.createFromText(comment.text as string)}
+            // text={comment.text}
+            uid={comment.uid}
+            createdAt={comment.created_at}
+            reactions={comment.reactions}
+            editComment={(text) => handleEditComment(comment.id, text)}
+        />
+    );
+
+    const handleOpenThread = React.useCallback((threadId: number) => {
+        setOpenThreadIds((curr) => new Set([...curr, threadId]));
+    }, []);
+
+    const renderThreadCommentDOM = (
+        parentCommentId: number,
+        comments: IComment[]
+    ) => {
+        if (openThreadIds.has(parentCommentId)) {
+            return comments.map(renderFlatCommentDOM);
+        }
+        const uids = Array.from(
+            new Set(comments.map((comment) => comment.uid))
+        );
+        return (
+            <div className="ClosedCommentThread flex-row">
+                <div className="flex-row mr8">
+                    {uids.map((uid) => (
+                        <UserAvatar
+                            key={`thread-avatar-${uid}-${parentCommentId}`}
+                            uid={uid}
+                            tiny
+                        />
+                    ))}
+                </div>
+                <StyledText
+                    className="ThreadCount mr8"
+                    size="xsmall"
+                    color="accent"
+                    cursor="default"
+                >
+                    {comments.length}{' '}
+                    {comments.length === 1 ? 'Reply' : 'Replies'}
+                </StyledText>
+                <span className="LastReplyDate">
+                    <StyledText size="xsmall" color="lightest" cursor="default">
+                        Last reply{' '}
+                        {fromNow(comments[comments.length - 1].updated_at)}
+                    </StyledText>
+                </span>
+                <span className="HoverText">
+                    <StyledText size="xsmall" color="lightest" cursor="default">
+                        View Thread
+                    </StyledText>
+                </span>
+            </div>
+        );
+    };
+
+    const renderCommentDOM = () =>
+        comments.map((comment: IComment) => {
+            if (comment.child_comments) {
+                return (
```

Comment:
can renderThreadCommentDOM take care of rendering flat and also handleOpenThread portion as well?
then this function can be reduced to a simple if else